### ISSUE TITLE
Provide more details when an Internal Server error occurs

### ISSUE THREAD

**Issue by pploegaert** [openvstorage/framework#676]
Autotest logs:

```
 ======================================================================
 ERROR: test suite for <module 'ci.tests.vdisk' from '/opt/OpenvStorage/ci/tests/vdisk/__init__.py'>
 ----------------------------------------------------------------------
 Traceback (most recent call last):
   File "/usr/lib/python2.7/dist-packages/nose/suite.py", line 208, in run
     self.setUp()
   File "/usr/lib/python2.7/dist-packages/nose/suite.py", line 291, in setUp
     self.setupContext(ancestor)
   File "/usr/lib/python2.7/dist-packages/nose/suite.py", line 314, in setupContext
     try_run(context, names)
   File "/usr/lib/python2.7/dist-packages/nose/util.py", line 471, in try_run
     return func()
   File "/opt/OpenvStorage/ci/tests/vdisk/__init__.py", line 39, in setup
     GeneralVPool.validate_vpool_sanity(expected_settings=vpool_params)
   File "/opt/OpenvStorage/ci/tests/general/general_vpool.py", line 259, in validate_vpool_sanity
     vpool_config = GeneralVPool.get_configuration(vpool)
   File "/opt/OpenvStorage/ci/tests/general/
 general_vpool.py", line 133, in get_configuration
     timeout=60)
   File "/opt/OpenvStorage/ci/tests/general/connection.py", line 157, in execute_get_action
     response = urllib2.urlopen(request).read()
   File "/usr/lib/python2.7/urllib2.py", line 127, in urlopen
     return _opener.open(url, data, timeout)
   File "/usr/lib/python2.7/urllib2.py", line 410, in open
     response = meth(req, response)
   File "/usr/lib/python2.7/urllib2.py", line 523, in http_response
     'http', request, response, code, msg, hdrs)
   File "/usr/lib/python2.7/urllib2.py", line 448, in error
     return self._call_chain(*args)
   File "/usr/lib/python2.7/urllib2.py", line 382, in _call_chain
     result = func(*args)
   File "/usr/lib/python2.7/urllib2.py", line 531, in http_error_default
     raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
 HTTPError: HTTP Error 500: INTERNAL SERVER ERROR
```

While actual exception in ovs-webapp-api.log shows:

```
2016-06-28 07:00:57 46500 +0200 - ovs54b - 22249/140199222572528 - api/middleware - 95 - ERROR - An unhandled exception occurred: 'dict' object has no attribute 'id'
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/django/core/handlers/base.py", line 112, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python2.7/dist-packages/rest_framework/viewsets.py", line 78, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/django/views/decorators/csrf.py", line 57, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/rest_framework/views.py", line 399, in dispatch
    response = self.handle_exception(exc)
  File "/usr/lib/python2.7/dist-packages/rest_framework/views.py", line 396, in dispatch
    response = handler(request, *args, **kwargs)
  File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 77, in new_function
    return f(*args, **kw)
  File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 315, in new_function
    return Response(task.id, status=status.HTTP_200_OK)
AttributeError: 'dict' object has no attribute 'id'
```

**Comment by khenderick** [comment]
Won't fix the actual bug (caused by bad backwards compatibility - but since the hypervisor removal is such a huge change we can't keep them backwards compatible anyway), but this is a request to raise a more meaningful error to the client.

The error (500, internal server error) in the above snippet is correct and should be kept but I highly suspect that the returned content is `<h1>Internal Server Error</h1>` because it's uncatched and it should instead be changed to sth like `{"error": "internal_server_error", error_description": "<actual error message>"}`, which can be handled by some middleware.

**Comment by khenderick** [comment]
Resolved by #690

**Comment by kinvaris** [comment]
PASSED:
- Stopped a arakoon cluster so it got no master anymore.
- Visited the backend page
- Copied a 500 request and executed it via `curl`:

```
curl 'https://10.100.199.161/api/alba/backends/06a4fc14-25f4-42a6-bc68-2e00c780c237/?contents=_dynamics%2C-ns_data%2C_relations&timestamp=1472821940996' -H 'DNT: 1' -H 'Accept-Encoding: gzip, deflate, sdch' -H 'GET: /api/vpools/?page=1&page_size=10&sort=name&contents=_dynamics%2Cbackend_type&timestamp=1471340986400' -H 'Accept-Language: nl,en-US;q=0.8,en;q=0.6' -H 'Authorization: Bearer i7a2dmSXhJhY3h!0K!=-{_2GZbQDbg]xJ7.uVg9ld9KUSa#KfWSHf[8Pk~CgJQgz' -H 'Content-Type: application/json' -H 'Accept: application/json; version=*' -H 'Referer: https://10.100.199.161/' -H 'X-Requested-With: XMLHttpRequest' -H 'Connection: keep-alive' -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/49.0.2623.110 Safari/537.36' --compressed --insecure
{"error_description": "Client_helper.MasterLookupResult.Error(0)", "error": "internal_server"}
```
